File: recursion/nested-arrays.py
def nestedAdd(array):
    sum = 0
    for i in range(0, len(array)):
      current = array[i]
      if type(array[i]) == type([]):
        sum += nestedAdd(current)
      else:
        sum += current
    return sum


print(nestedAdd([12, 3]))

File: recursion/nested-arrays.js
var total = 0;

function nestedAdd(array) {
    let sum = 0;
    for (let i = 0; i < array.length; i++) {
      const current = array[i];
      if (Array.isArray(current)) {
        sum += nestedAdd(current);
      } else {
        sum += current;
      }
    }
    return sum;
}

console.log(nestedAdd([1, 2, [3, 1]]))
